h/static/scripts/test/root-thread-test.js

```diff
@@ -179,6 +185,47 @@ describe('rootThread', function () {
     ]);
   });
 
+  describe('when the thread filter query is set', function () {
+    it('generates a thread filter function to match annotations', function () {
+      fakeBuildThread.reset();
+
+      fakeAnnotationUI.state = Object.assign({}, fakeAnnotationUI.state,
+        {selectedTab: 'annotation'});
+
+      rootThread.thread(fakeAnnotationUI.state);
+      var threadFilterFn = fakeBuildThread.args[0][1].threadFilterFn;
+
+      var annotation = {target: [{ selector: {} }]};
+      assert(threadFilterFn({annotation: annotation}));
```

Comment:
@sheetaluk - I would strongly 👍 this comment. Debugging failures of `assert.predicate(a, b)` is _much_ easier than `assert(condition)`